adr/ADR-52.md

```diff
@@ -0,0 +1,123 @@
+# JetStream Read-after-Write
+
+| Metadata | Value                                                              |
+|----------|--------------------------------------------------------------------|
+| Date     | 2025-07-11                                                         |
+| Author   | @MauriceVanVeen                                                    |
+| Status   | Proposed                                                           |
+| Tags     | jetstream, kv, objectstore, server, client, spec, refinement, 2.12 |
```

Comment:
`spec` is for big giant things like KV so can probably drop it here - refinements inherently wont be specs